Pei_HW/session03/wsgi/calculator.py

```diff
@@ -0,0 +1,92 @@
+#!/usr/bin/env python3
+def add (x, y):
+    return x + y
+def subtract (x, y):
+    return x - y
+def multiply (x, y):
+    return x * y
+def divide (x, y):
+    return x / y
+
+def page():
+    front_page = """<html>
+    <head>
+    <title>Simple Calculator</title>
+    </head>
+
+    <body>
+    <p>This simple calculator can perform the following math related operations:</p>
+    <p>Addition</p>
+    <p>ex: http://localhost:8080/add/5/3</p>
+    <p>Subtraction</p>
+    <p>ex: http://localhost:8080/subtract/5/3</p>
+    <p>Multiplication</p>
+    <p>ex: http://localhost:8080/multiply/5/3</p>
+    <p>Division</p>
+    <p>ex: http://localhost:8080/divide/5/3</p>
+    </body>
+    </html>"""
+    return front_page
+
+def calculate (operator, operands):
+    answer_page = """<html>
+    <head>
+    <title>Simple Calculator</title>
+    </head>
+    <body>
+    Answer: {}
+    </body>
+    </html>"""
+    if operator == 'add':
+        answer = add(*operands)
+    elif operator == 'subtract':
+        answer = subtract (*operands)
+    elif operator == 'multiply':
+        answer = multiply(*operands)
+    elif operator == 'divide':
+        answer = divide(*operands)
+    else:
+        raise ValueError
+    return answer_page.format(answer)

```

Comment:
Because you are raising a ValueError, and not including a message no error will show up on your page. You should either include a message:
<pre> raise ValueError("Please use 'add', 'subtract', 'multiply', or 'divide'")</pre>
or catch the ValueError in your application function, and address it there. Also, it shouldn't be a ValueError, but probably a NameError, since these are not numbers.